lib/gpw/client.rb

```diff
@@ -2,36 +2,101 @@
 
 module GPW
   class Client
+    def initialize(client: Faraday.new)
+      @client = client
+    end
+
     def fetch_companies_list
-      response = fetch_gpw_stock_list
+      response = fetch_stock_list
       parse_response(response)
     end
 
-    private
+    def fetch_gpw_companies_details
+      response = fetch_gpw_stock_list
+      parse_money_response_gpw(response)
+    end
 
-    def fetch_gpw_stock_list
-      conn_gpw = Faraday.new(
-        url: 'https://gpw.notoria.pl/widgets/ta/symbols.php',
-        headers: {
-          'Accept' => 'application/json',
-          'Referer' => 'https://gpw.notoria.pl',
-          'DNT' => '1',
-          'Sec-GPC' => '1',
-        },
-      )
+    def fetch_nc_companies_details
+      response = fetch_nc_stock_list
+      parse_money_response_nc(response)
+    end
 
-      conn_gpw.get
+    private
+
+    def fetch_stock_list
+      @client.get do |req|
```

Comment:
I think I showed you another syntax some time ago and you preferred it:
```ruby
@client.get(url: '...', headers : {...})
```

I also like it more.

Reply:
have you checked the faraday documentation? there are examples of this syntax for sure

Reply:
![obraz](https://user-images.githubusercontent.com/6394528/162042338-bd8cdae4-d684-4d2b-9e85-3b3d3941f158.png)
https://lostisland.github.io/faraday/usage/

Reply:
That is strange, I read exactly this documentation, and much more, and I try to set this for many ways, and this good way as well, it wasn't work. Then I try today and it works from start :) malice of inanimate objects..